code/modules/mob/living/carbon/xenomorph/castes/king/abilities_conqueror.dm

```diff
@@ -0,0 +1,979 @@
+/datum/action/ability/xeno_action/conqueror_endurance/proc/adjust_particles()
+	var/particles_dir = closest_cardinal_dir(xeno_owner.dir)
+	particle_holder.particles.icon_state = "[xeno_owner.xeno_caste.caste_name] Blocking[particles_dir]"
+	particle_holder.layer = xeno_owner.layer + 0.01
+	particle_holder.pixel_y = xeno_owner.pixel_y + 16
+	particle_holder.pixel_x = xeno_owner.pixel_x + 32
+	particle_holder.alpha = initial(particle_holder.alpha)
+	animate(particle_holder, alpha = 0, time = 0.2 SECONDS)
+
+/particles/conqueror_endurance
+	icon = 'icons/effects/particles/conqueror.dmi'
+	width = 96
+	height = 96
+	count = 1
+	spawning = 1
+	lifespan = 1
+
+
+// ***************************************
+// *********** Conqueror's Domination
+// ***************************************
+#define CONQUEROR_DOMINATION_CASTING_DELAY 1.5 SECONDS // The wind-up delay for this ability.
+#define CONQUEROR_DOMINATION_MAX_PUSH_RANGE 3 // How far can our ability knock back its victims.
+
+/datum/action/ability/activable/xeno/conqueror_domination
+	name = "Domination"
+	desc = "Teleport towards a target location, distorting reality, and creating powerful shockwaves upon reappearing."
+	action_icon = 'icons/Xeno/actions/wraith.dmi'
+	action_icon_state = "Banish"
+	ability_cost = 150
+	cooldown_duration = 30 SECONDS
+	use_state_flags = ABILITY_USE_BUCKLED
+	target_flags = ABILITY_TURF_TARGET
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_CONQUEROR_DOMINATION,
+	)
+	/// The casting range for this ability (in tiles).
+	var/casting_range = 5
+	/// The area of effect range for this ability (in tiles).
+	var/effect_range = 2
+
+/datum/action/ability/activable/xeno/conqueror_domination/on_cooldown_finish()
+	. = ..()
+	xeno_owner.playsound_local(xeno_owner, 'sound/effects/alien/new_larva.ogg', 35, 0)
+	xeno_owner.balloon_alert(xeno_owner, "[initial(name)] ready")
+
+/datum/action/ability/activable/xeno/conqueror_domination/can_use_action(...)
+	. = ..()
+	if(!.)
+		return
+	if(!xeno_owner.canmove)
+		return FALSE
+	return TRUE
+
+/datum/action/ability/activable/xeno/conqueror_domination/use_ability(atom/atom_target)
+	var/turf/turf_target = isturf(atom_target) ? atom_target : atom_target.loc
+	if(isclosedturf(turf_target) || isspaceturf(turf_target) || isspacearea(get_area(turf_target)))
+		xeno_owner.balloon_alert(xeno_owner, "Cannot go there")
+		return
+	if(!line_of_sight(xeno_owner, turf_target) || IS_OPAQUE_TURF(turf_target))
+		xeno_owner.balloon_alert(xeno_owner, "No vision")
+		return
+	var/turf/current_turf = xeno_owner.loc
+	var/turf/turf_to_check = current_turf
+	var/check_distance = min(casting_range, get_dist(xeno_owner, turf_target))
+	var/list/valid_turfs = list()
+	for(var/x = 1 to check_distance)
+		turf_to_check = get_step(current_turf, get_dir(current_turf, turf_target))
+		if(!turf_to_check)
+			break
+		for(var/atom/atom_to_check AS in turf_to_check)
+			if(!atom_to_check.CanPass(xeno_owner, turf_to_check))
+				break
```

Comment:
can this use get_line for this?

Reply:
It sure can! I've changed this, and a repeat case in the Dash ability, in the next commit.